view/rooms/utils/newRoom.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * rooms/utils/newRoom.js
+ */
+
+const utils = require('../../utils.js');
+const ZERO = 0;
+
+module.exports = {
+  /**
+   * Checks if a room already exists with a given name.
+   *
+   * @param {String} name - Name of the room to look for.
+   * @returns {Promise} - Resolves to true if room exists, false otherwise
+   */
+  checkIfRoomExistsFromName(name) {
+    const getUrl = `/v1/rooms?name=${name}`;
+    return utils.getPromiseWithUrl(getUrl)
+    .then((res) => {
+      if (res.body && res.body[ZERO]) {
+        return true;
+      }
+
+      return false;
+    });
+  },
+
+  /**
+   * Creates a room from parameters
+   *
+   * @param {Object} paramaters - Parameters to use when creating the room
+   * @returns {Promise} - Resolves when room is created.
+   */
+  createRoomFromParameters(paramaters) {
+    const postUrl = '/v1/rooms';
+    const room = {
```

Comment:
Is there any validation when the parameter has no attributes as expected?

Reply:
@cassiodias there is no validation on the frontend yet but the backend will not create the room if it does not have a name or a type. Also, this function will only be called if it has these fields (ref: https://github.com/salesforce/refocus/pull/986/files#diff-8ea0877b79ba6bad7603e84e1bf270a3R91)